src/lib.rs

```diff
@@ -1900,18 +1901,21 @@ impl Bindings {
     }
 
     /// Gets the rustfmt path to rustfmt the generated bindings.
-    fn rustfmt_path<'a>(&'a self) -> io::Result<Cow<'a, PathBuf>> {
+    fn rustfmt_path<'a>(&'a self) -> io::Result<Option<Cow<'a, PathBuf>>> {
         debug_assert!(self.options.rustfmt_bindings);
         if let Some(ref p) = self.options.rustfmt_path {
-            return Ok(Cow::Borrowed(p));
+            return Ok(Some(Cow::Borrowed(p)));
         }
         if let Ok(rustfmt) = env::var("RUSTFMT") {
-            return Ok(Cow::Owned(rustfmt.into()));
+            return Ok(Some(Cow::Owned(rustfmt.into())));
         }
+        #[cfg(feature = "which-rustfmt")]
         match which::which("rustfmt") {
-            Ok(p) => Ok(Cow::Owned(p)),
+            Ok(p) => Ok(Some(Cow::Owned(p))),
             Err(e) => Err(io::Error::new(io::ErrorKind::Other, format!("{}", e))),
         }
+        #[cfg(not(feature = "which-rustfmt"))]
```

Comment:
I'd just return an error here, like `Err(io::Error::new(io::ErrorKind::Other, "which wasn't enabled, and no rustfmt binary specified")))`, then the patch becomes much simpler, wdyt?

The error is not fatal so you'd still generate bindings.